### DISCUSSION TITLE
Webpack+Asciidoctor-kroki: Cannot statically analyse 'require(…, …)'

### DISCUSSION THREAD

**Discussion by lord-pendragon** [webpack/webpack#15136] [Q&A]
![image](https://user-images.githubusercontent.com/97427464/148748098-ea9434bc-c3fa-4816-92f1-6a16fecdd504.png)

Description: Using Asciidoctor's Kroki extension with Webpackv4 results in the following error
The author of Asciidoctor-Kroki was approached, the error seems to be beyond the scope of their package.

How to Reproduce:

    
    npx create-react-app asciidoctor-kroki-react-scripts
    cd asciidoctor-kroki-react-scripts
    npm i @asciidoctor/core@2.2.5 asciidoctor-kroki@0.15.4
    
Edit src/App.js, add the following import:
import Asciidoctor from '@asciidoctor/core'
import kroki from 'asciidoctor-kroki'
Save and run

     npm start

Repo: https://github.com/lord-pendragon/ascii-krokii-webpack-issue

![image](https://user-images.githubusercontent.com/97427464/148748391-21eda0a5-db93-4f67-8314-b6a419eb4f65.png)

**Comment by alexander-akait** [comment]
Something broken in CRA, please report them, also I recommend to migrate on new version with webpack v5

**Comment by jschulte-xtl** [comment]
**BUMP, as this is STILL an ongoing Issue!**

I've the same problem while using NestJS with Webpack within a Monorepo and it boils down to this problem:

- https://stackoverflow.com/questions/69898244/cannot-statically-analyse-require-in-line-16

which also https://gist.github.com/tchetwin analyses here:

- https://gist.github.com/tchetwin/eeeffdfbbf973b59f05a3ecba3ab253a

and yes, there does not seem to be a solution as of now... except changing the library itself...

PS: Also opened an Issue within the `deep-clone` library as it also happens there. See:

https://github.com/jonschlinkert/clone-deep/issues/30
